Automattic/woocommerce-payments
# feat: add mechanism to manage UPE flag

Fixes https://github.com/Automattic/woocommerce-payments/issues/2178
Helps with https://github.com/Automattic/woocommerce-payments/issues/2169

#### Changes proposed in this Pull Request









Creating a mechanism to enable/disable the UPE feature flag.
This "mechanism" uses the ReactJS Context API.
To demonstrate its functionality, I am leveraging the "UPE payment methods" onboarding task: https://github.com/Automattic/woocommerce-payments/issues/2169

This mechanism will be used in multiple places.

#### Testing instructions





- Ensure that [in your DB](http://localhost:8083/index.php?db=wordpress&table=wp_options&target=sql.php), do `DELETE FROM wp_options WHERE option_name='wcpay_additional_methods_setup_completed'`
- Ensure you have the latest version of WCPay Dev tools `cd docker/wordpress/wp-content/plugins/woocommerce-payments-dev-tools && git pull`
- In WCPay Dev tools, ensure the "Enable UPE checkout" checkbox is not checked
- In WCPay Dev tools, with the "Enable UPE settings changes" checkbox not checked, no task is displayed in WC Home
- in WCPay Dev tools, with the "Enable UPE settings changes" and "Add UPE additional payment methods" checkboxes checked, a new task is displayed in WC Home
![Screen Shot 2021-06-16 at 6 20 52 PM](https://user-images.githubusercontent.com/273592/122307257-9b0ffb00-cecf-11eb-8064-8bf837124589.png)
- Click on the task
- Clicking "enable" enables UPE (you can confirm in WCPay Dev as well, the "Enable UPE checkout" checkbox should now be checked)

![2021-06-16 18 24 17](https://user-images.githubusercontent.com/273592/122307526-24bfc880-ced0-11eb-92da-aa3cdec8f07d.gif)


-------------------
